src/runtime-rs/crates/resource/src/cpu_mem/mem.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2019-2023 Alibaba Cloud
+// Copyright (c) 2019-2023 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use anyhow::{Context, Ok, Result};
+use hypervisor::Hypervisor;
+use kata_types::config::TomlConfig;
+use oci::LinuxResources;
+use tokio::sync::RwLock;
+
+use crate::cpu_mem::initial_size::InitialSizeManager;
+use crate::ResourceUpdateOp;
+
+// MIB_TO_BYTES_SHIFT the number to shift needed to convert MiB to Bytes
+pub const MIB_TO_BYTES_SHIFT: i32 = 20;
+
+#[derive(Default, Debug, Clone)]
+pub struct MemResource {
+    /// Current memory
+    pub(crate) current_mem: Arc<RwLock<u32>>,
```

Comment:
do we really need this Arc<RwLock<u32>> field which is only used to judge whether to clear balloon size or not?

Reply:
As runtime is a multi-thread application, there might be a situation that we have one thread that needs to update the current_mem,  while other threads are reading it to make decisions based on the current state

Reply:
I'm not focus on the Arc Rwlock, I mean this field is never used again after you refactor balloon logic.

Reply:
fixed